Gifski/Preview/createAVAssetFromGIF.swift

```diff
@@ -0,0 +1,175 @@
+	let pixelBufferAdaptor = AVAssetWriterInputPixelBufferAdaptor(
+		assetWriterInput: writerInput,
+		sourcePixelBufferAttributes: [
+			kCVPixelBufferPixelFormatTypeKey as String: kCVPixelFormatType_32ARGB,
+			kCVPixelBufferWidthKey as String: firstCGImage.width,
+			kCVPixelBufferHeightKey as String: firstCGImage.height,
+			kCVPixelBufferCGImageCompatibilityKey as String: true,
+			kCVPixelBufferCGBitmapContextCompatibilityKey as String: true
+		]
+	)
+	guard assetWriter.canAdd(writerInput) else {
+		throw CreateAVAssetError.cannotAddWriterInput
+	}
+	assetWriter.add(writerInput)
+
+	guard assetWriter.startWriting() else {
+		throw CreateAVAssetError.failedToStartWriting
+	}
+	assetWriter.startSession(atSourceTime: .zero)
+
+	var frameIndex = 0
+
+	let frameRate: CMTimeScale = await {
+		if let settingFrameRate = settings.conversion.frameRate {
+			return CMTimeScale(settingFrameRate)
+		}
+		if let inputFrameRate = try? await settings.conversion.asset.frameRate {
+			return CMTimeScale(inputFrameRate)
+		}
+		return CMTimeScale(30.0)
+	}()
+
+
+	let dispatchQueue = DispatchQueue(label: "com.gifski.assetWriterQueue")
+	let sendableImageSource = SendableWrapper(imageSource)
+	let sendableWriterInput = SendableWrapper(writerInput)
+	let sendablePixelBufferAdaptor = SendableWrapper(pixelBufferAdaptor)
+
+	try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<Void, any Error>) in
+		writerInput.requestMediaDataWhenReady(on: dispatchQueue) {
+			do {
+				var progressThreshold = 0.05
+				while sendableWriterInput.value.isReadyForMoreMediaData && frameIndex < numberOfImages {
+					try Task.checkCancellation()
+					let progress = Double(frameIndex) / Double(numberOfImages)
+					if progress > progressThreshold {
+						Task {
+							await onProgress(progress)
+						}
+						progressThreshold = progress + 0.05
+					}
+					defer {
+						frameIndex += 1
+					}
+					guard let cgImage = sendableImageSource.value.createImage(atIndex: frameIndex) else {
+						throw CreateAVAssetError.failedToCreateImage
+					}
+					guard let pixelBufferPool = sendablePixelBufferAdaptor.value.pixelBufferPool else {
+						throw CreateAVAssetError.failedToCreatePixelBufferPool
+					}
+					let pixelBuffer = try createPixelBuffer(from: cgImage, using: pixelBufferPool)
+
+					let presentationTime = CMTime(
+						value: CMTimeValue(frameIndex),
+						timescale: frameRate
+					)
+					guard sendablePixelBufferAdaptor.value.append(pixelBuffer, withPresentationTime: presentationTime) else {
+						throw CreateAVAssetError.cannotAppendNewImage
+					}
+				}
+				if frameIndex >= numberOfImages {
+					sendableWriterInput.value.markAsFinished()
+					continuation.resume()
+				}
+			} catch {
+				sendableWriterInput.value.markAsFinished()
+				continuation.resume(throwing: error)
+			}
+		}
+	}
+	await assetWriter.finishWriting()
+	guard assetWriter.status != .failed else {
+		throw CreateAVAssetError.failedToWrite
+	}
```

Comment:
Shouldn't this be:

```suggestion
	guard assetWriter.status == .completed else {
		throw (assetWriter.status.error ?? CreateAVAssetError.failedToWrite)
	}
```

currently it ignores `.cancelled` and `.unknown`.